templates/tools/dockerfile/gcp_api_libraries.include

```diff
@@ -1,2 +1,2 @@
 # Google Cloud platform API libraries
-RUN pip install --upgrade google-api-python-client oauth2client
+RUN pip install --upgrade google-auth==1.24.0 google-api-python-client==1.12.8 oauth2client==4.1.0
```

Comment:
If we're pinning dependencies, we want to do it for the entire transitive dependency closure. Typically, I would do this as follows:

```
pip install ${FIRST_ORDER_DEPENDENCIES}
pip freeze > requirements.lock
```

Then, in the Dockerfile, include `pip install -r requirements.lock`

Reply:
I explored this idea a bit. It's a bit tricky that this requirements file need to be in the build scope of Docker. I guess the better plan is to create a unified-neat-locked requirements file for all CI jobs, as another PR.

Reply:
It's kind of a trade-off here. It's easier to install Python packages in scripts, but it saves cpu cycles if we do it in Docker images.

Reply:
My thought was that you do the `pip freeze` on your dev machine and then just paste the results into the Dockerfile.

Reply:
Also, you could use a heredoc or something like that to embed `requirements.lock` into the Dockerfile.

Reply:
The requirements.lock is pretty long. It would be tricky to manage across Dockerfiles. I think one of the points of having requirements file is allowing source control with one ground truth config. I will try to create a common requirements.lock for our CI jobs, and update our `push_testing_images.sh` to be able to fetch the requirements file while building Docker images.

(BTW, this is a backport to v1.36.x to enable interop images build)